File: examples/Fortran/rutherford_boeing/rb_read.f90
! examples/Fortran/rutherford_boeing/rb_read.f90
! Example code for SPRAL_RUTHERFORD_BOEING
program rb_read_example
   use spral_rutherford_boeing
   use spral_matrix_util, only: print_matrix, &
                                SPRAL_MATRIX_REAL_SYM_INDEF
   implicit none

   ! Parameters
   integer, parameter :: long = selected_int_kind(18)
   integer, parameter :: wp = kind(0d0)

   ! Matrix data
   character(len=72) :: title
   integer :: m, n
   integer(long), dimension(:), allocatable :: ptr
   integer, dimension(:), allocatable :: row
   real(wp), dimension(:), allocatable :: val

   ! rb_read options and inform
   type(rb_read_options) :: options
   integer :: inform

   ! Read matrix
   call rb_read("matrix.rb", m, n, ptr, row, val, options, inform, &
      title=title)

   ! Print matrix
   write(*, "(3a)") "Matrix '", trim(title), "'"
   call print_matrix(6, -1, SPRAL_MATRIX_REAL_SYM_INDEF, m, n, ptr, row, val)

end program rb_read_example

File: examples/C/rutherford_boeing/rb_read.c
/* examples/C/rutherford_boeing/rb_read.c
 * Example code for SPRAL_RUTHERFORD_BOEING */
#include <stdint.h>
#include <stdio.h>
#include "spral.h"

int main(void) {
   /* Matrix data */
   void *handle;
   char title[73]; // Maximum length 72 character + '\0'
   int matrix_type, m, n;
   int64_t *ptr;
   int *row;
   double *val;

   /* Initalise options */
   struct spral_rb_read_options options;
   spral_rb_default_read_options(&options);

   /* Read matrix */
   spral_rb_read("matrix.rb", &handle, &matrix_type, &m, &n, &ptr, &row, &val,
         &options, title, NULL, NULL);

   /* Print matrix */
   printf("Matrix '%s'\n", title);
   spral_print_matrix_i64d(-1, matrix_type, m, n, ptr, row, val, 0);

   /* Free handle */
   spral_rb_free_handle(&handle);
}
